src/controller/ProductController/buildQuery.ts

```diff
@@ -52,6 +54,14 @@ export const buildQuery = (req: Request) => {
     })
   }
 
+  if (req.query?.limit) {
+    query.size = req.query.limit
```

Comment:
should we do some sanity checking here? what if query.limit is a string? Or 1014881481834747482821714744?